test/groovy/NeoDeployTest.groovy

```diff
@@ -122,6 +132,107 @@ class NeoDeployTest extends BasePiperTest {
                 .hasSingleQuotedOption('source', archiveName))
     }
 
+    @Test
+    void extensionsAsStringTest() {
+
+        def checkedExtensionFiles = []
+
+        StepAssertions.metaClass.static.assertFileExists =
+            { Script step, String filePath ->
+                checkedExtensionFiles << filePath
+                if( ! [archiveName, 'myExtension.yml'].contains(filePath) )
+                    step.error("File ${filePath} cannot be found.")
+            }
+
+        stepRule.step.neoDeploy(
+                script: nullScript,
+                source: archiveName,
+                mtaExtensionDescriptors: 'myExtension.yml'
+        )
+
+        assert checkedExtensionFiles.contains('myExtension.yml')
+
+        assertThat(shellRule.shell,
+            new CommandLineMatcher()
+                .hasProlog('neo.sh deploy-mta')
+                .hasSingleQuotedOption('extensions', 'myExtension.yml'))
+    }
+
+    @Test
+    void extensionsAsSetTest() {
+        Set extensions= ['myExtension1.yml' ,'myExtension2.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+
+    @Test
+    void extensionsAsListTest() {
+        List extensions= ['myExtension1.yml' ,'myExtension2.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+    @Test
+    void sameExtensionProvidedTwiceTest() {
+        List extensions= ['myExtension1.yml' ,'myExtension2.yml', 'myExtension1.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+
+    void extensionsAsCollectionTest(def extensions) {
+
+        def checkedExtensionFiles = []
+
+        StepAssertions.metaClass.static.assertFileExists =
+            { Script step, String filePath ->
+                checkedExtensionFiles << filePath
+                if( ! [
+                    archiveName,
+                    'myExtension1.yml',
+                    'myExtension2.yml',
+                    ].contains(filePath) )
+                    step.error("File ${filePath} cannot be found.")
+            }
+
+        stepRule.step.neoDeploy(
+                script: nullScript,
+                source: archiveName,
+                mtaExtensionDescriptors: extensions
+        )
+
+        assert checkedExtensionFiles.contains('myExtension1.yml')
+        assert checkedExtensionFiles.contains('myExtension2.yml')
+
+        assertThat(shellRule.shell,
+            new CommandLineMatcher()
+                .hasProlog('neo.sh deploy-mta')
+                // some kind of creative usage for the single quotation check (... single quotes inside)
+                .hasSingleQuotedOption('extensions', 'myExtension1.yml\',\'myExtension2.yml'))
+
+    }
+
+    @Test
+    void extensionsForWrongDeployModeTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage('Extensions are only supported for deploy mode \'MTA\'')
+
+        StepAssertions.metaClass.static.assertFileExists =
```

Comment:
This is not needed

Reply:
@alejandraferreirovidal : yes, you are right. The check for a deplomode-extensions mismatch happens earlier like the file exists check. --> removed